app/pulp/app/models/auth.py

```diff
@@ -0,0 +1,65 @@
+"""
+Django models related to the Authentication system
+"""
+from gettext import gettext as _
+
+from django.core import validators
+from django.contrib.auth.models import BaseUserManager, AbstractBaseUser, PermissionsMixin
+from django.db import models
+from django.utils import timezone
+
+
+class PulpUserManager(BaseUserManager):
+    """
+    A custom manager for PulpUser objects.
+    """
+    use_in_migrations = True
+
+    def _create_user(self, username, password, is_superuser):
+        """
+        Creates and saves a PulpUser with the given username and password.
+        """
+        now = timezone.now()
+        if not username:
+            raise ValueError('The given username must be set')
+        user = self.model(username=username, is_active=True, is_superuser=is_superuser,
+                          date_joined=now)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, username, password=None):
+        return self._create_user(username, password, False)
+
+    def create_superuser(self, username, password):
+        return self._create_user(username, password, True)
+
+
+class PulpUser(AbstractBaseUser, PermissionsMixin):
+    """
+    A custom Django User class for Pulp.
+    """
+    username = models.CharField(_('username'), max_length=150, unique=True,
+        help_text=_('Required. 150 characters or fewer. Letters, digits and '
+                    '@/./+/-/_ only.'),
```

Comment:
In other models we did not specified the `help_text` in models but we did it later in serializers. For consistency (and maybe for the API docs which may use `help_text` from serializers if I am not mistaken) I suggest to move it to serializer. What do you think?